### PR TITLE
accfft: added new package

### PR DESCRIPTION
**Pull request by alalazo**
The shared build depends on https://github.com/amirgholami/accfft/pull/13 to work. 

@amirgholami If you could tag a release, I would gladly add it here with a version number and a proper checksum :wink: 

@ursache You might be interested in this package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/accfft/package.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        'step3', 'step3f',
+        'step4',
+        'step5', 'step5f',
+    ]
+
+    def cmake_args(self):
+        spec = self.spec
+        return [
+            '-DFFTW_ROOT={0}'.format(spec['fftw'].prefix),
+            '-DFFTW_USE_STATIC_LIBS=false ',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/accfft/package.py]: U probably don’t need space

File: var/spack/repos/builtin/packages/accfft/package.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Accfft(CMakePackage):
```
Comment (davydden) [inline var/spack/repos/builtin/packages/accfft/package.py]: Optionally one could also derive from CudaPackage, then u will have ‘cuda_arch’ variant.

Comment (alalazo) [reply var/spack/repos/builtin/packages/accfft/package.py]: Thanks!

File: var/spack/repos/builtin/packages/accfft/package.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        'accfft', 'accfft_utils',
+        'step1', 'step1f',
+        'step2', 'step2f',
+        'step3', 'step3f',
+        'step4',
+        'step5', 'step5f',
+    ]
+
+    def cmake_args(self):
+        spec = self.spec
```
Comment (davydden) [inline var/spack/repos/builtin/packages/accfft/package.py]: something like
```
cuda_arch = spec.variants['cuda_arch'].value
if cuda_arch is not None:
   # '-arch=sm_{0}'.format(cuda_arch[0])
```

Comment (alalazo) [reply var/spack/repos/builtin/packages/accfft/package.py]: @davydden  I'll check tomorrow how this package works with CMake CUDA, thanks for the hints

**Comment by alalazo** [comment]
As soon as https://github.com/amirgholami/accfft/pull/14 is done, this PR can be merged as well

**Comment by alalazo** [comment]
Thanks for the review @davydden !

**Pull request merged**